include/net/ieee802154_radio.h

```diff
@@ -168,11 +169,42 @@ enum ieee802154_config_type {
 	/** Sets the current MAC frame counter value for radios supporting transmit security. */
 	IEEE802154_CONFIG_FRAME_COUNTER,
 
-	/** Configure a radio reception slot */
+	/** Configure a radio reception slot. This can be used for any scheduler reception, e.g.:
+	 *  Zigbee GP device, CSL, TSCH, etc.
+	 *
+	 *  In order to configure a CSL receiver the upper layer should combine several
+	 *  configuration options in the following way:
+	 *    1. Use ``IEEE802154_CONFIG_ENH_ACK_HEADER_IE`` once to inform the radio driver of the
+	 *  short and extended addresses of the peer to which it should inject CSL IEs.
+	 *    2. Use ``IEEE802154_CONFIG_CSL_RX_TIME`` periodically, before each use of
+	 *  ``IEEE802154_CONFIG_CSL_PERIOD`` setting parameters of the nearest CSL RX window, and
+	 *  before each use of IEEE_CONFIG_RX_SLOT setting parameters of the following (not the
+	 *  nearest one) CSL RX window, to allow the radio driver to calculate the proper CSL Phase
+	 *  to the nearest CSL window to inject in the CSL IEs for both transmitted data and ack
+	 *  frames.
+	 *    3. Use ``IEEE802154_CONFIG_CSL_PERIOD`` on each value change to update the current CSL
+	 *  period value which will be injected in the CSL IEs together with the CSL Phase based on
+	 *  ``IEEE802154_CONFIG_CSL_RX_TIME``.
+	 *    4. Use ``IEEE802154_CONFIG_RX_SLOT`` periodically to schedule the immediate receive
+	 *  window earlier enough before the expected window start time, taking into account
+	 *  possible clock drifts and scheduling uncertainties.
+	 *
+	 *  This diagram shows the usage of the four options over time:
+	 *        Start CSL                                  Schedule CSL window
+	 *
+	 *    ENH_ACK_HEADER_IE                           CSL_RX_TIME (following window)
+	 *         |                                        |
+	 *         | CSL_RX_TIME (nearest window)           | RX_SLOT (nearest window)
+	 *         |    |                                   |   |
+	 *         |    | CSL_PERIOD                        |   |
+	 *         |    |    |                              |   |
+	 *         v    v    v                              v   v
+	 *    ----------------------------------------------------------[ CSL window ]-----
```

Comment:
It was just a sketch :laughing:

Reply:
But good enough to better understand the description :)